src/VisualStudio/Core/Def/Implementation/ExtractClass/VisualStudioExtractClassOptionsService.cs

```diff
@@ -0,0 +1,127 @@
+                        IsCheckable = true
+                    });
+
+            using var cancellationTokenSource = new CancellationTokenSource();
+            var memberToDependentsMap = SymbolDependentsBuilder.FindMemberToDependentsMap(membersInType, document.Project, cancellationTokenSource.Token);
+
+            var conflictingTypeNames = selectedType.ContainingNamespace.GetAllTypes(cancellationTokenSource.Token).Select(t => t.Name);
+            var candidateName = selectedType.Name + "Base";
+            var defaultTypeName = NameGenerator.GenerateUniqueName(candidateName, name => !conflictingTypeNames.Contains(name));
+
+            var containingNamespaceDisplay = selectedType.ContainingNamespace.IsGlobalNamespace
+                ? string.Empty
+                : selectedType.ContainingNamespace.ToDisplayString();
+
+            var generatedNameTypeParameterSuffix = GetTypeParameterSuffix(document, selectedType);
+
+            var viewModel = new ExtractClassViewModel(
+                _waitIndicator,
+                notificationService,
+                memberViewModels,
+                memberToDependentsMap,
+                defaultTypeName,
+                containingNamespaceDisplay,
+                document.Project.Language,
+                generatedNameTypeParameterSuffix,
+                conflictingTypeNames.ToImmutableArray(),
+                document.GetRequiredLanguageService<ISyntaxFactsService>());
+
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+            var dialog = new ExtractClassDialog(viewModel);
+
+            var result = dialog.ShowModal();
+
+            if (result.GetValueOrDefault())
+            {
+                return new ExtractClassOptions(
+                    viewModel.DestinationViewModel.FileName,
+                    viewModel.DestinationViewModel.TypeName,
+                    viewModel.DestinationViewModel.Destination == CommonControls.NewTypeDestination.CurrentFile,
+                    viewModel.MemberSelectionViewModel.CheckedMembers.SelectAsArray(m => new ExtractClassMemberAnalysisResult(m.Symbol, m.MakeAbstract)));
+            }
+
+            return null;
+        }
+
+        private string GetTypeParameterSuffix(Document document, INamedTypeSymbol type)
+        {
+            var typeParameters = type.TypeParameters;
+
+            if (type.TypeParameters.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            var displayParts = type.ToDisplayParts();
+            var typeParameterNames = displayParts.Where(d => d.Kind == SymbolDisplayPartKind.TypeParameterName).SelectAsArray(d => d.ToString());
+
+            var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+            return Formatter.Format(syntaxGenerator.SyntaxGeneratorInternal.TypeParameterList(typeParameterNames), document.Project.Solution.Workspace).ToString();
```

Comment:
whatever is going on here seems super hacky.

Reply:
🤷‍♂️ This fell into the category of shared code moving around. If there's a better way to get the "<T>" of the display I'm down. I tried refactoring to use display parts but it wasn't clear that would even work

Reply:
That should be `<T>`